src/runtime/RegExpObject.cpp

```diff
@@ -204,51 +204,47 @@ bool RegExpObject::defineOwnProperty(ExecutionState& state, const ObjectProperty
     return returnValue;
 }
 
+bool RegExpObject::isRegExpObject(ExecutionState& state)
+{
+    Object* proto = state.context()->globalObject()->regexpPrototype();
+    return this == proto || this->getPrototypeObject(state) == proto;
```

Comment:
I think this is could be wrong
because there are multiple global objects exist in real usage(ex. iframe)
thus, This situation {variable `proto` is regexpPrototype of another globalObject.} can be happened
How about derive new class from Object for testing this? (class RegexpObjectProtoype { bool isRegexpPrototype...(); }
The above thing(deriving new class) is one of my suggestion

Reply:
Multiple global objects should not be a problem there, since each global object should follow the same rules. The `RegExp` and the `RegExp.prototype` objects are immutable bindings in the global lexical environment, so it cannot be overwritten or be deleted but can be extended. This code part only depends on the _valid_ existence of the `RegExp.prototype` object, so it cannot cause any problem.

Reply:
this is detailed description of problem.
If user makes multiple context, ex) iframe
you can see there is multiple globalObject instances.
the problem is user can call cross-context function
```
eg) 
iframe->
function someFunctionInIframe() {
   // testing RegExpObject::isRegExpObject(ExecutionState& state) and failed
}
iframe owner->
document.querySelector("iframe").contentWindow.someFunctionInIframe();
```
globalObject could differ <regexp object in iframe> and outer window global object
because ExecutionState is copied from caller's ExecutionState
and ExecutionState firstly comes from Script::execute(<outer window context>)

Reply:
Ohh, after that if I'm not mistaken the only problem is the pointer equality check for the `this` and the `RegExp.prototype` object, isn't it?

Reply:
Yes.. regexp object where comes from and state.globaObject is could differ

Reply:
Okay, I think we can easily fix this problem with a virtual `isRegExpPrototype()` function. Thanks for the review, that was a good catch.